sites/shared/components/sets/measie-input.mjs

```diff
@@ -0,0 +1,209 @@
+import { isDegreeMeasurement } from 'config/measurements.mjs'
+import { measurementAsMm, formatMm } from 'shared/utils.mjs'
+import { Collapse } from 'shared/components/collapse.mjs'
+import { PlusIcon, EditIcon } from 'shared/components/icons.mjs'
+import { useState } from 'react'
+export const ns = ['account']
+
+const Mval = ({ m, val = false, imperial = false, className = '' }) =>
+  val ? (
+    isDegreeMeasurement(m) ? (
+      <span className={className}>{val}°</span>
+    ) : (
+      <span
+        dangerouslySetInnerHTML={{ __html: formatMm(val, imperial ? 'imperial' : 'metric') }}
+        className={className}
+      />
+    )
+  ) : null
+
+const heightClasses = {
+  2: 'h-12',
+  4: 'h-10',
+  8: 'h-8',
+  16: 'h-6',
+  32: 'h-4',
+}
+const fractionClasses =
+  'w-full border-2 border-solid border-base-100 hover:border-secondary bg-secondary rounded bg-opacity-50 hover:bg-opacity-100'
+const FractionButtons = ({ t, fraction }) => (
+  <div className="flex flex-row mt-1 content-center items-center justify-around">
+    <span className="text-xs inline-block pr-2">{t('fractions')}</span>
+    <div className="grow max-w-2xl flex items-baseline">
+      {Array.from({ length: 31 }, (_null, i) => {
+        let denom = 32
+        let num = i + 1
+
+        for (var n = 4; n > 0; n--) {
+          let fac = Math.pow(2, n)
+          if (num % fac === 0) {
+            denom = 32 / fac
+            num = num / fac
+            break
+          }
+        }
+
+        return (
+          <span className="group w-[3.125%] relative" key={i}>
+            <button
+              className={`${heightClasses[denom]} ${fractionClasses}`}
+              title={`${num}/${denom}"`}
+              onClick={() => fraction(num, denom)}
+            />
+            <span className="group-hover:visible invisible text-xs text-center absolute left-0 -bottom-6">{`${num}/${denom}"`}</span>
+          </span>
+        )
+      })}
+    </div>
+  </div>
+)
+export const MeasieRow = (props) => {
+  const { t, m, mset } = props
+
+  const isSet = typeof mset.measies?.[m] === 'undefined' ? false : true
+
+  return (
+    <Collapse
+      color="secondary"
+      openTitle={t(m)}
+      title={
+        <>
+          <div className="grow text-left md:text-right block md:inline font-bold pr-4">{t(m)}</div>
+          {isSet ? (
+            <Mval m={m} val={mset.measies[m]} imperial={mset.imperial} className="w-1/3" />
+          ) : (
+            <div className="w-1/3" />
+          )}
+        </>
+      }
+      toggle={isSet ? <EditIcon /> : <PlusIcon />}
+      toggleClasses={`btn ${isSet ? 'btn-secondary' : 'btn-neutral bg-opacity-50'}`}
+    >
+      <MeasieInput {...props} />
+    </Collapse>
+  )
+}
+
+export const MeasieInput = ({
+  t,
+  m,
+  mset,
+  startLoading,
```

Comment:
I think it makes more sense to let this default to `() => null` rather than pass in `emptyFunc` later?

Reply:
Some for `stopLoading` obvs